src/app/core/rpc/address-rpc.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { Log } from 'ng2-logger';
+import { Observer } from 'rxjs/Observer';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/observable/throw';
+
+import { RPCService } from './rpc.service';
+import { AddressCount } from './models/address-count.model';
+import { Address } from './models/address.model';
+import { Http } from '@angular/http';
+import {ElectronService} from 'ngx-electron';
+
+@Injectable()
+export class AddressRpcService {
+
+  log: any = Log.create('address-rpc.service');
+
+  private API_URL: string = '';
+
+  /**
+   *
+   * @param http
+   * @param rpc
+   */
+  constructor(private http: Http, public rpc: RPCService) {
+    this.API_URL = this.rpc.getApiUrl();
+  }
+
+  public getAddressCount(): Observable<AddressCount> {
+    const postData = this.rpc.getPostData('filteraddresses', [-1]);
+    const options = this.rpc.getRequestOptions();
+    return this.http
+      .post(this.API_URL, postData, options )
+      .map(response => {
+        const addressCount = response.json().result;
+        this.log.d('getAddressCount, addressCount:', addressCount);
+        return new AddressCount(addressCount);
+      })
+      .catch(this.handleError);
+  }
+
+  public findAddresses(params: Array<any>): Observable<Address[]> {
+    const postData = this.rpc.getPostData('filteraddresses', params);
+    const options = this.rpc.getRequestOptions();
+    return this.http
+      .post(this.API_URL, postData, options )
```

Comment:
My initial idea was to have a generic BaseRPCService which the AddressService would extend, and which would contain all the generic and common functionalities, but I couldn't figure out yet how to implement that. Also, in that case I should have splitted the AddressRpcService to AddressRpcService and AddressCountRpcService, which I didn't want to do yet.